### PR TITLE
add debug/trace logging for table creation/deletion

### PR DESCRIPTION
**Pull request by tjwilson90**
## General
**Before this PR**:
Table creation/deletion and how created table names are mapped to internal names is difficult to observe.

**After this PR**:
Table creation/deletion is logged at debug. When trace logging is enabled, a stacktrace is also logged.

**Priority**:
P2, needed in service of a long standing P2 ticket.

**Concerns / possible downsides (what feedback would you like?)**:
None

**Is documentation needed?**:
No

## Compatibility
**Does this PR create any API breaks (e.g. at the Java or HTTP layers) - if so, do we have compatibility?**:
No

**Does this PR change the persisted format of any data - if so, do we have forward and backward compatibility?**:
No

**The code in this PR may be part of a blue-green deploy. Can upgrades from previous versions safely coexist? (Consider restarts of blue or green nodes.)**:
Yes

**Does this PR rely on statements being true about other products at a deployment - if so, do we have correct product dependencies on these products (or other ways of verifying that these statements are true)?**:
No

**Does this PR need a schema migration?**
No

## Testing and Correctness
**What, if any, assumptions are made about the current state of the world? If they change over time, how will we find out?**:
None

**What was existing testing like? What have you done to improve it?**:
N/A

**If this PR contains complex concurrent or asynchronous code, is it correct? The onus is on the PR writer to demonstrate this.**:
N/A

**If this PR involves acquiring locks or other shared resources, how do we ensure that these are always released?**:
N/A

## Execution
**How would I tell this PR works in production? (Metrics, logs, etc.)**:
Turn on the logging.

**Has the safety of all log arguments been decided correctly?**:
This should be checked.

**Will this change significantly affect our spending on metrics or logs?**:
By default, no.

**How would I tell that this PR does not work in production? (monitors, etc.)**:
Turning on the logging.

**If this PR does not work as expected, how do I fix that state? Would rollback be straightforward?**:
Easy rollback.

**If the above plan is more complex than “recall and rollback”, please tag the support PoC here (if it is the end of the week, tag both the current and next PoC)**:

## Scale
**Would this PR be expected to pose a risk at scale? Think of the shopping product at our largest stack.**:
No

**Would this PR be expected to perform a large number of database calls, and/or expensive database calls (e.g., row range scans, concurrent CAS)?**:
No

**Would this PR ever, with time and scale, become the wrong thing to do - and if so, how would we know that we need to do something differently?**:
No

## Development Process
**Where should we start reviewing?**:

**If this PR is in excess of 500 lines excluding versions lock-files, why does it not make sense to split it?**:

**Please tag any other people who should be aware of this PR**:
@jeremyk-91
@sverma30
@raiju

### CODE REVIEW

File: atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleDdlTable.java [hunk 1]
```diff
@@ -590,4 +607,12 @@ private SqlConnection getCompactionConnection() {
 
         return sqlConnection;
     }
+
+    private static void logDebugOrTrace(@CompileTimeConstant String message, Arg<?>... args) {
+        if (log.isTraceEnabled()) {
+            log.trace(message, Arrays.asList(args), new SafeRuntimeException("provided for stack trace"));
+        } else if (log.isDebugEnabled()) {
+            log.debug(message, Arrays.asList(args));
```
Comment (barisoyoruk) [inline atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleDdlTable.java:615]: What if both are enabled? Is it okay not to log at the debug level? One may be unaware that trace is enabled and try to see the logs at the debug level.

**Comment by barisoyoruk** [comment]
Can we use `LoggingArgs` for everything instead of `UnsafeArg.of`?

**Review by jeremyk-91 — COMMENTED** [review]
The change makes sense. That said, had a couple of questions around log safety:

- Maybe @CVdV-au or @berler can advise more precisely on this: I see you've marked the physical table names in `OracleDdlTable` and `PostgresDdlTable` as unsafe, though I've thought I've seen places internally where they're treated as safe (and I know that for Cassandra we definitely treat this as safe).

**Comment by berler** [comment]
> The change makes sense. That said, had a couple of questions around log safety:
> 
> * Maybe @CVdV-au or @berler can advise more precisely on this: I see you've marked the physical table names in `OracleDdlTable` and `PostgresDdlTable` as unsafe, though I've thought I've seen places internally where they're treated as safe (and I know that for Cassandra we definitely treat this as safe).

Whether these are safe or not depend on the table mapper you are actually using. Our internal product does in fact have this always be safe due to mapping all table names (regardless of what they are) to a number. The problem is the default table mapper doesn't do this and instead the physical table name contains a prefix of the user-provided table name. So for anything that isn't our internal product, these actually shouldn't be considered safe (unless you know for sure that the user-provided table name is already safe). We have many products that dynamically create table names with names coming directly from actual user input, so you cannot consider these to be safe.

File: atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleDdlTable.java [hunk 2]
```diff
@@ -419,6 +432,10 @@ private void dropTableInternal(String fullTableName, String shortTableName) {
                             fullTableName);
             oracleTableNameGetter.clearCacheForTable(fullTableName);
         }
+        logDebugOrTrace(
+                "dropped oracle table",
+                UnsafeArg.of("fullTableName", fullTableName),
+                UnsafeArg.of("shortTableName", shortTableName));
```
Comment (jeremyk-91) [inline atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleDdlTable.java:438]: discussed offline: this is correct

**Review by jeremyk-91 — APPROVED** [review]
Thanks for the context. Yep, ok - so at this layer we don't know, and so Unsafe is correct for the full table names (but the Short ones are of course `_n_1234567890` and so fine).

**Pull request merged**